frontend/src/components/smart/MerchandiseList.vue

```diff
@@ -132,36 +117,41 @@ export default Vue.extend({
       for (const product of this.products) {
         product.price = await this.getItemPrice({id: product.id});
       }
-      console.log(this.products);
     },
 
     async fetchCountries() {
       const response = await axios.get('http://localhost:2400/countries');
       this.countries = response.data?.result;
+      this.fetchStates();
+    },
+
+    fetchStates() {
+      this.recipient.stateCode = undefined;
+      this.states = this.countries.find((country: Country) => country.code === this.recipient.countryCode)?.states || [];
     },
 
-    async openAddress(product: Product) {
-      this.$refs['merchandise-address-modal'].show();
-      this.selected_product = product;
+    openAddress(product: Product) {
+      (this.$refs['merchandise-address-modal'] as any).show();
+      this.selectedProduct = product;
     },
 
     async buyItem() {
-      if(!this.selected_product) return;
+      if(!this.selectedProduct) return;
       const printful_payload = {
         recipient: this.recipient,
         items: [
           {
-            sync_variant_id: this.selected_product.id,
+            sync_variant_id: this.selectedProduct.id,
             quantity: 1
           }
         ]
       };
 
-      if(!this.selected_product.price) return;
-      await this.purchaseMerchandise({id: this.selected_product.id, price: this.selected_product.price, amount: 1});
+      if(!this.selectedProduct.price) return;
+      await this.purchaseMerchandise({id: this.selectedProduct.id, price: this.selectedProduct.price, amount: 1});
+      this.recipient = {} as Recipient;
```

Comment:
Do we want to clear recipient after every order? That would mean users have to refill the form from scratch, correct me if I'm wrong.